modules/redpanda/redpanda.go

```diff
@@ -0,0 +1,308 @@
+package redpanda
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+	"os"
+	"text/template"
+	"time"
+
+	"github.com/docker/go-connections/nat"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/api/admin"
```

Comment:
usually, modules should not be couple to a client library.

Reply:
This is just a HTTP client. I could write the HTTP calls on my own, but I don't think that would be better. The HTTP call is needed to create a service account when creating a Redpanda module with authentication enabled.